c/tests/test_tables.c

```diff
@@ -105,6 +105,104 @@ insert_edge_metadata(tsk_table_collection_t *tables)
+    tsk_size_t example_metadata_length = (tsk_size_t) strlen(example_metadata);
+    tsk_size_t example_metadata_schema_length
+        = (tsk_size_t) strlen(example_metadata_schema);
+
+    // Test equality empty tables
+    ret = tsk_table_collection_init(&tc1, 0);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_init(&tc2, 0);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_equivalent(
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
+    CU_ASSERT_TRUE(ret);
+
+    // Adding some meat to the tables
+    ret = tsk_node_table_add_row(&tc1.nodes, TSK_NODE_IS_SAMPLE, 0.0, 0, 0, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_node_table_add_row(&tc1.nodes, TSK_NODE_IS_SAMPLE, 1.0, 0, 0, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_individual_table_add_row(&tc1.individuals, 0, NULL, 0, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_population_table_add_row(&tc1.populations, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_edge_table_add_row(&tc1.edges, 0.0, 1.0, 1, 0, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_site_table_add_row(&tc1.sites, 0.2, "A", 1, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_mutation_table_add_row(
+        &tc1.mutations, 0, 0, TSK_NULL, TSK_UNKNOWN_TIME, NULL, 0, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+
+    // Equality of empty vs non-empty
+    ret = tsk_table_collection_equivalent(
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
+    CU_ASSERT_FALSE(ret);
+    // Copying over tc1 to tc2
+    ret = tsk_table_collection_copy(&tc1, &tc2, 0);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+
+    // equivalent for different metadata/schema
+    ret = tsk_table_collection_set_metadata(
+        &tc1, example_metadata, example_metadata_length);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_equivalent(&tc1, &tc2, 0);
+    CU_ASSERT_TRUE(ret);
+    ret = tsk_table_collection_equivalent(
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
+    CU_ASSERT_FALSE(ret);
+    ret = tsk_table_collection_set_metadata(
+        &tc2, example_metadata, example_metadata_length);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_equivalent(
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
+    CU_ASSERT_TRUE(ret);
+    ret = tsk_table_collection_set_metadata_schema(
+        &tc1, example_metadata_schema, example_metadata_schema_length);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_equivalent(&tc1, &tc2, 0);
+    CU_ASSERT_TRUE(ret);
+    ret = tsk_table_collection_equivalent(
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
+    CU_ASSERT_FALSE(ret);
+    ret = tsk_table_collection_equivalent(&tc1, &tc2, 0);
+    CU_ASSERT_TRUE(ret);
+    ret = tsk_table_collection_set_metadata_schema(
+        &tc2, example_metadata_schema, example_metadata_schema_length);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
```

Comment:
You can leave out the fatal here and in the other equality tests below and above - it's more informative for when the tests fail to carry on and get output for the other cases you're testing. Generally `FATAL` is used for "something has gone _so_ wrong there's no point continuing"

Reply:
thanks for the explanation, Ben! fixed.